plugins/isecl/client/isecl-plugin.go

```diff
@@ -0,0 +1,459 @@
+	keyTransferUrl := km.config.Kbs.URL() + "/keys/" + keyID + "/dhsm2-transfer"
+	headers := map[string]string{
+		"Accept":           "application/json",
+		"Accept-Challenge": "SGX",
+		"Authorization":    "Bearer " + km.config.Kbs.BearerToken,
+	}
+	if len(sessionID) != 0 {
+		headers["Session-Id"] = sessionID
+	}
+
+	km.log.Info("Sending key transfer request", "url", keyTransferUrl, "session", sessionID)
+	return km.client.Get(keyTransferUrl, headers)
+}
+
+func (km *iSeclPlugin) openSessionForKeyTransfer(keyID string, quote []byte, publicKey []byte, nonce []byte) (string, []byte, error) {
+	response, status, err := km.initiateKeyTransfer(keyID, "")
+	if err != nil {
+		return "", nil, fmt.Errorf("post request for keys failed: %v", err)
+	}
+	if status != http.StatusUnauthorized {
+		// Expected an unauthorized access as no session
+		// key transfer still exists
+		return "", nil, fmt.Errorf("unexpected response (code %d) from key server: %v", status, string(response))
+	}
+	challenge := kbs.ChallengeRequest{}
+	if err := json.Unmarshal(response, &challenge); err != nil {
+		return "", nil, fmt.Errorf("failed to parse response: %v", err)
+	}
+	km.log.Info("Key transfer response", "challenge", challenge)
+	// If session creation successful, use decoded challenge as sessionID
+	id, err := base64.StdEncoding.DecodeString(challenge.Challenge)
+	if err != nil {
+		return "", nil, fmt.Errorf("failed to decode session id from challenge: %v", err)
+	}
+	// NOTE(avalluri): For KBS session, there is no way to pass the 'nonce'.
+	// It only expects quote and the public key.
+	swk, err := km.createSession(quote, publicKey, challenge.Challenge)
+	if err != nil {
+		return "", nil, err
+	}
+	session := challenge.ChallengeType + ":" + string(id)
+
+	return session, swk, nil
+}
+
+// getWrappedKey fetches wrapped SWK and Private key from the KBS server
+func (km *iSeclPlugin) getWrappedKey(keyID string, quote []byte, publicKey []byte, nonce []byte) ([]byte, []byte, error) {
+	sessionID, swk, err := km.openSessionForKeyTransfer(keyID, quote, publicKey, nonce)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to initiate key transfer: %v", err)
+	}
+
+	response, status, err := km.initiateKeyTransfer(keyID, sessionID)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to re-initiate key transfer: %v", err)
+	}
+	if status != http.StatusOK {
+		return nil, nil, fmt.Errorf("unexpected response (code %d) from the server: %s", status, string(response))
+	}
+	result := kbs.KeyTransferResponse{}
+	if err := json.Unmarshal(response, &result); err != nil {
+		return nil, nil, fmt.Errorf("failed to unmarshal key transfer response: %v", err)
+	}
+
+	wrappedKey, err := base64.StdEncoding.DecodeString(result.KeyInfo.KeyData)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to base64 decode the wrapped key returned by the server: %v", err)
+	}
+	km.log.Info("wrappedKey", "data", wrappedKey, "len(before truncate)", len(wrappedKey))
```

Comment:
I guess that because this is wrapped, it's not too bad to write the key to the logs, but should we not print the wrapped key data just in case?

Reply:
This log was used for debugging in the development. This is not needed anymore and removed.